### PR TITLE
feat(ci): integrate E2E tests from feature file to Xray

### PR DESCRIPTION
**Pull request by a-eljazouly**
## Description
This GitHub Action (GHA) integrates an end-to-end (E2E) test, described in a .feature file, with Xray. When a feature file is added or modified in a development branch, this action automatically creates a test ticket in Xray.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x
- [x] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

To verify that the goal of this PR is met, follow these steps:

1. Add or modify a .feature file in the development branch.
2. Observe the automatic creation of a test ticket in Xray.

**Note**: Ensure that Xray / Jira configuration is correctly set up to receive test tickets .

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/cypress/e2e/Test-Xray/test2.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#targetVersions:Cloud components:centreon-web testSet:MON-21774
```
Comment (bdauria) [inline centreon/tests/e2e/cypress/e2e/Test-Xray/test2.feature:1]: targetVersions should be automatically implied from the branch (develop actually being both develop and Coud).

Components is also implied from the project. centreon => centreon-web

File: .github/workflows/create_jira_test.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: Integrate e2e tests to Xray
+
+on:
+  push:
+    branches:
+      - develop
+      - dev-23.04.x
+      - dev-22.10.x
+      - dev-22.04.x
+      - dev-21.10.x
```
Comment (kduret) [inline .github/workflows/create_jira_test.yml:10]:
```suggestion
    branches:
      - develop
      - dev-[2-9][0-9].[0-9][0-9].x
```

File: .github/workflows/create_jira_test.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    with:
+      version_file: centreon/www/install/insertBaseConf.sql
+
+  run_script:
+    needs: [get-version]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
```
Comment (kduret) [inline .github/workflows/create_jira_test.yml:24]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/create_jira_test.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      - dev-22.04.x
+      - dev-21.10.x
+        
+jobs:
+  get-version:
+    uses: ./.github/workflows/get-version.yml
+    with:
+      version_file: centreon/www/install/insertBaseConf.sql
+
+  run_script:
```
Comment (kduret) [inline .github/workflows/create_jira_test.yml:18]:
```suggestion
  synchronize-jira-xray:
```

File: .github/workflows/create_jira_test.yml
Comment (kduret) [inline .github/workflows/create_jira_test.yml:1]: `synchronize-jira-xray` seems more appropriate for workflow name

File: .github/scripts/create_jira_test.py
Comment (kduret) [inline .github/scripts/create_jira_test.py:1]: `synchronize_jira_xray.py` seems more appropriate for script name

File: .github/workflows/create_jira_test.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          echo "CHANGED_FILES=${CHANGED_FILES[@]}" >> $GITHUB_ENV
+          echo "Changed .feature files: ${CHANGED_FILES[@]}"
+
+      - name: Run Script on Changed Feature Files
+        if: env.CHANGED_FILES != ''
+        run: |
+          echo "The target version for this script is : ${{ needs.get-version.outputs.major_version }}"
+          for file in $CHANGED_FILES; do
+            echo "Running script for $file on branch ${{ github.ref_name }}"
+            python .github/scripts/create_jira_test.py "$file" "${{ github.ref_name }}" "${{ needs.get-version.outputs.major_version }}"
```
Comment (kduret) [inline .github/workflows/create_jira_test.yml:54]:
```suggestion
            python .github/scripts/synchronize_jira_xray.py "$file" "${{ github.ref_name }}" "${{ needs.get-version.outputs.major_version }}"
```

**Review by kduret — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**